str0m/src/session.rs

```diff
@@ -207,6 +217,11 @@ impl Session {
                 }
             }
         }
+
+        let queue_states: Vec<_> = only_m_line(&self.m_lines)
+            .map(|line| line.buffers_tx_queue_state(now))
+            .collect();
```

Comment:
This is a heap allocation on every handle_timeout – something I tried to avoid when possible.

I haven't looked into how Rust optimizes this, but since the m-line count is pretty static, I wonder if we should at least move the vec to be a reused instance variable on Session?

Reply:
I've been thinking the same thing, but have been ignoring it as a premature optimisation for the time being. Can fix though via instance variable on Session before merging, if you wish

Reply:
Addressed in 9dd30d2